ngraph/core/src/op/space_to_depth.cpp

```diff
@@ -46,43 +47,13 @@ std::shared_ptr<Node> ov::op::v0::SpaceToDepth::clone_with_new_inputs(const Outp
 
 void ngraph::op::v0::SpaceToDepth::validate_and_infer_types() {
     NGRAPH_OP_SCOPE(v0_SpaceToDepth_validate_and_infer_types);
-    ov::PartialShape data_pshape = get_input_partial_shape(0);
 
     const auto& data_type = get_input_element_type(0);
-
-    auto data = input_value(0);
-
-    if (data_pshape.is_static()) {
-        const auto& data_shape = data.get_shape();
-
-        NODE_VALIDATION_CHECK(this,
-                              !(data_shape.size() < 3),
-                              "The input tensor with rank lower than 3 is not supported (input rank: ",
-                              data_shape.size(),
-                              ")");
-
-        auto multiplier = std::pow(m_blocksize, data_shape.size() - 2);
-
-        auto out_shape = data_shape;
-        out_shape[1] *= multiplier;
-        for (size_t i = 2; i < out_shape.size(); i++) {
-            NODE_VALIDATION_CHECK(this,
-                                  m_blocksize > 0 && !(out_shape[i] % m_blocksize),
-                                  "The dimension on position: ",
-                                  i,
-                                  " equal to: ",
-                                  out_shape[i],
-                                  " must be a multiple of m_blocksize: ",
-                                  m_blocksize);
-
-            out_shape[i] /= m_blocksize;
-        }
-
-        set_output_size(1);
-        set_output_type(0, data_type, out_shape);
-    } else {
-        set_output_type(0, data_type, ov::PartialShape::dynamic(data_pshape.rank()));
-    }
+    std::vector<ov::PartialShape> output_shapes = {ov::PartialShape{}};
+    const std::vector<ov::PartialShape> input_shapes = {get_input_partial_shape(0)};
+    shape_infer(this, input_shapes, output_shapes);
+    set_output_size(1);
```

Comment:
I believe you could skip setting output size in all of the operations in this PR